src/main/kotlin/com/getkeepsafe/dexcount/DexCountTask.kt

```diff
@@ -18,44 +18,62 @@ package com.getkeepsafe.dexcount
 
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
+import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.FileCollection
+import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.logging.LogLevel
-import org.gradle.api.provider.Provider
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 import java.io.File
 import javax.annotation.Nullable
+import javax.inject.Inject
 
 /**
  * The maximum number of method refs and field refs allowed in a single Dex
  * file.
  */
 const val MAX_DEX_REFS: Int = 0xFFFF // 65535
 
-open class DexCountTask: DefaultTask() {
+@Suppress("UnstableApiUsage")
+open class DexCountTask @Inject constructor(
+    objectFactory: ObjectFactory
+): DefaultTask() {
     lateinit var tree: PackageTree
 
-    lateinit var inputFileProvider: () -> File
-
     /**
      * The output of the 'package' task; will be either an APK or an AAR.
      */
-    @InputFile
-    fun getInputFile(): File {
-        return inputFileProvider()
+    private fun getInputFile(): File {
+        return inputFileProperty.asFile.get()
     }
 
+    @InputFile
+    val inputFileProperty: RegularFileProperty = objectFactory.fileProperty()
+
     private val rawInputRepresentation: String
         get() = "${getInputFile()}"
 
+    @Input
     lateinit var variantOutputName: String
 
     @Nullable
```

Comment:
Why annotate @Nullable in Kotlin?

Reply:
Possibly an ancient holdover from Groovy?  Or possibly it's meaningful to Gradle's task input system?  I don't quite remember.